novawallet/Modules/WalletsList/Common/WalletIconView.swift

```diff
@@ -0,0 +1,80 @@
+import UIKit
+import SnapKit
+
+final class WalletIconView: UIView {
+    enum Constants {
+        static let holeWidth: CGFloat = 3
+        static let iconSize = CGSize(width: 32, height: 32)
+        static let networkIconSize = CGSize(width: 14, height: 14)
+        static let networkIconOffset = CGPoint(x: 6, y: 4)
+        static let radius: CGFloat = 4.5
+    }
+
+    let iconViewImageView = UIImageView()
+    let networkIconImageView = UIImageView()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        setupLayout()
+    }
+
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        guard !networkIconImageView.isHidden else {
+            return
+        }
+
+        iconViewImageView.layoutIfNeeded()
+
+        let width = networkIconImageView.bounds.width + Constants.holeWidth * 2
```

Comment:
I believe we have this calculation I would add another method to UIView where ```cutHole``` resides to handle this